File: camelCase4/Java/CamelCase.java
public class CamelCase {
    public static void main(String[] args) {
//            Scanner scanner = new Scanner(System.in);
//            List<String> tests = new ArrayList<>();
//
//            while (scanner.hasNext()) {
//                tests.add(scanner.nextLine());
//            }
//
//            scanner.close();
//
//            for (String input : tests) {
//                formatter(input);
//            }

    }

    public static void formatter(String input) {
        if (input.startsWith("S")) {
            if (input.endsWith("()")) input = input.replace("()", "");
            StringBuilder result = new StringBuilder(input.substring(4));
            for (int i = 1; i < result.length(); i++) {
                if (Character.isUpperCase(result.charAt(i))) {
                    CharSequence space = " ";
                    result.insert(i, space);
                    i++;
                }
            }
            System.out.println(result.toString().toLowerCase());
        } else {
            char method = input.charAt(2);
            String[] sentence = input.substring(4).split(" ");

            String result = "";

            for (String word : sentence) {
                word = word.substring(0, 1).toUpperCase() + word.substring(1);
                result += word;
            }



            switch (method) {
                case 'C':
                    System.out.println(result);
                    break;
                case 'V':
                    System.out.println(result.substring(0, 1).toLowerCase() +
                            result.substring(1));
                    break;
                case 'M':
                    System.out.println(result.substring(0, 1).toLowerCase() +
                            result.substring(1) + "()");
                    break;
                default:
                    System.out.println("hello!");
            }

        }
    }

}

File: camelCase4/JS/camelCase.js
function processData(input) {
  const tests = input.split('\n');
  
  for (let sentence of tests) {
     if (sentence.startsWith('S')) {
          const letters = sentence.replace('()', '').substring(4).split('');
          let result = '';
          for (let l of letters) {
              const regex = new RegExp('[A-Z]');
              if (regex.test(l)) {
                  l = ' ' + l.toLowerCase();
                  result += l;
              } else {
                  result += l;
              }
              
          }
          console.log(result.trim());
      } else {
          const method = sentence[2];
          const words = sentence
                          .substring(4)
                          .split(' ')
                          .map(w => w.substring(0, 1).toUpperCase() + w.substring(1));
          const clas = words.join('').trim();
          const variable = clas.substring(0, 1).toLowerCase() + clas.substring(1);
          const meth = variable + "()";
          
          switch (method) {
              case 'C':
                  console.log(clas);
                  break;
              case 'V':
                  console.log(variable);
                  break;
              case 'M':
                  console.log(meth);
                  break;
              default:
                  console.log("no!");
          }
      }
   
  }

} 

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
  _input += input;
});

process.stdin.on("end", function () {
 processData(_input);
});
